src/app/api/client.ts

```diff
@@ -43,18 +43,31 @@ export function createApiClient<T extends Record<string, any>>(
   return createClient<T>({
     baseUrl: `${baseUrl}${path ?? ''}`,
     fetch: fetchRetry(fetch, {
-      retryOn: [
-        StatusCodes.TOO_MANY_REQUESTS, // Retry also when concurrency limits (due to external factors) are hit
-        StatusCodes.BAD_GATEWAY,
-        StatusCodes.SERVICE_UNAVAILABLE,
-        StatusCodes.CONFLICT,
-        StatusCodes.GATEWAY_TIMEOUT,
-        StatusCodes.REQUEST_TIMEOUT,
-        StatusCodes.INTERNAL_SERVER_ERROR,
-      ],
+      retryOn: (_, __, response) => {
+        // Concurrency limits (due to external factors) are hit
+        if (response?.status === StatusCodes.TOO_MANY_REQUESTS) {
+          const resetDuration = response?.headers.get('ratelimit-reset');
```

Comment:
When I was thinking about this, I thought the `x-ratelimit-remaining` is best header for this. It's value is 0 if ratelimit is hit and is larger than zero when daily limit is hit.

But I don't know if you discussed it with Tomas.

Reply:
Yes, I did, and this is exactly what I suggested, but he's reworking rate limit headers [here](https://github.com/i-am-bee/bee-api/pull/134), and the proper remaining count will be sent even for this case, so using `ratelimit-reset` is his suggestion.